src/main/java/net/io/kino/service/impl/ShowtimeServiceImpl.java

```diff
@@ -0,0 +1,64 @@
+package net.io.kino.service.impl;
+
+import net.io.kino.model.Movie;
+import net.io.kino.model.Showroom;
+import net.io.kino.model.Showtime;
+import net.io.kino.repository.ShowtimeRepository;
+import net.io.kino.service.ShowtimeService;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.time.LocalDateTime;
+import java.util.List;
+
+public class ShowtimeServiceImpl implements ShowtimeService {
+
+    @Autowired
+    private ShowtimeRepository showtimes;
+
+    @Override
+    public Showtime createShowtime(Showtime showtime) {
+
+        List < Showtime > existingShowtimes = showtimes.findShowtimesByShowroom(showtime.getShowroom());
+
+        for (Showtime existingShowtime: existingShowtimes) {
+            if (isShowtimeSameDayThatAnother(showtime.getDate(), existingShowtime)) {
+                if (isShowtimeDuringAnother(showtime, existingShowtime) ||
+                        willAnotherShowtimeStartDuringThis(showtime, existingShowtime) ||
+                        willThisShowtimeStartDuringAnother(showtime, existingShowtime)){
+                    throw new IllegalArgumentException("This showtime overlaps with another showtime.");
+                }
+            }
+        }
+        return showtimes.save(showtime);
+    }
+
+    @Override
+    public Showtime getShowtimeById(long id) { return (Showtime) showtimes.findShowtimeById(id); }
```

Comment:
Nielegalne castowanie Listy na pojedynczy element
Tutaj też można użyć findById żeby zwrócić jeden showtime, bo id jest unikalne to zawsze zwróci jeden element